### ISSUE TITLE
Mat(Mat) construction

### ISSUE THREAD

**Issue by schmrlng** [SimonDanisch/FixedSizeArrays.jl#65]
``` julia
julia> using FixedSizeArrays

julia> a = Mat((1,2), (3,4))
FixedSizeArrays.Mat{2,2,Int64}(
    1 3
    2 4
)

julia> Mat(a)
FixedSizeArrays.Mat{4,1,Int64}(
    1
    1
    1
    1
)
```

I think this last call should return a copy of `a`. I suspect that `return :( $FSA($a_expr...) )` [here](https://github.com/SimonDanisch/FixedSizeArrays.jl/blob/master/src/constructors.jl#L30) should instead look something like `return :( $FSA($a_expr.(1)...) )`, but making that change introduces other test errors that are tricky to track down through the constructor magic. A likely related issue:

```
julia> Mat([1,2,3,4])  # Array{Float64,1}
FixedSizeArrays.Mat{4,1,Int64}(
    1
    1
    1
    1
)

julia> Mat([1,2,3,4]'')  # Array{Float64,2}
FixedSizeArrays.Mat{4,1,Int64}(
    1
    2
    3
    4
)
```

**Comment by SimonDanisch** [comment]
Okay, that's definitely a bug... Constructor magic is really just an euphomism for bad code :D 
I'll look into it...

**Comment by SimonDanisch** [comment]
should be fixed by #67
